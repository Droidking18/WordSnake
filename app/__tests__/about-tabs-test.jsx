// __tests__/About-test.jsx

import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent } from '@testing-library/react-native';

import About from '../(tabs)/about';

jest.mock('@expo/vector-icons/Ionicons', () => 'Ionicons');
jest.mock('react-native/Libraries/Image/Image', () => 'Image');
jest.mock('@/components/Collapsible', () => ({ Collapsible: 'Collapsible' }));
jest.mock('@/components/ExternalLink', () => 'ExternalLink');
jest.mock('@/components/ParallaxScrollView', () => 'ParallaxScrollView');
jest.mock('@/components/ThemedText', () => ({ ThemedText: 'ThemedText' }));
jest.mock('@/components/ThemedView', () => ({ ThemedView: 'ThemedView' }));

describe('About Screen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<About />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the header image', () => {
    const tree = renderer.create(<About />);
    const instance = tree.root;
    const headerImage = instance.findAllByType('Image');
    expect(headerImage).toBeTruthy();
  });

  it('contains the correct title', () => {
    const tree = renderer.create(<About />);
    const instance = tree.root;
    const titleText = instance.findByProps({ children: 'WordSnake' });
    expect(titleText).toBeTruthy();
  });

  it('contains the collapsible sections', () => {
    const tree = renderer.create(<About />);
    const instance = tree.root;
    const collapsibleSections = instance.findAllByType('Collapsible');
    expect(collapsibleSections.length).toBe(4);
  });
});
