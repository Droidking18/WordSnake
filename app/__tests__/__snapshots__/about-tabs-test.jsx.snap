// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Screen renders correctly 1`] = `
<ParallaxScrollView
  headerBackgroundColor={
    {
      "dark": "#353636",
      "light": "#ffffff",
    }
  }
  headerImage={
    <Image
      source={1}
      style={
        {
          "color": "#ffffff",
          "height": "100%",
          "position": "absolute",
          "resizeMode": "contain",
          "width": "100%",
        }
      }
    />
  }
>
  <ThemedView
    style={
      {
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <ThemedText
      type="title"
    >
      WordSnake
    </ThemedText>
  </ThemedView>
  <ThemedText>
    Instructions to play the game:
  </ThemedText>
  <Collapsible
    title="What's the goal?"
  >
    <ThemedText>
      The goal is simple! You need to form words by navigating the snake. When you have a word, you can lock it in by pressing the \`claim your word\` button. The longer the word, the more points you get!
    </ThemedText>
  </Collapsible>
  <Collapsible
    title="What if I collected in invalid word?"
  >
    <ThemedText>
      That's not a problem! You can always press the \`drop word\` button to start a new word, you get to keep your previous points, but as a penalty, your snake will keep the same length, as if you had collected the points.
    </ThemedText>
  </Collapsible>
  <Collapsible
    title="I want to earn more points!"
  >
    <ThemedText>
      Yay! You can earn more points by forming longer words. The longer the word, the more points you get! Another way to earn more points is by adjusting the difficulty levels. The higher the difficulty, the more points you get! There are many types of difficulty levels, the speed of the snake, and the number of letters. You can adjust these settings in the settings tab.
    </ThemedText>
  </Collapsible>
  <Collapsible
    title="How does the leader board work?"
  >
    <ThemedText>
      The leader board is a list of the top players in the game. You can see your ranking and the ranking of other players. Note WordSnake does not require you to log in, just select a name in the leader tab, and you're good to go! Note, you aren't required to use the leader board, but it's a fun way to see how you compare to other players.
    </ThemedText>
  </Collapsible>
</ParallaxScrollView>
`;
